domain/src/Commands/DomainCommands.php

```diff
@@ -0,0 +1,1288 @@
+      throw new DomainCommandException(dt('Nobody\'s listening on domain !d: (!e)', [
+        '!d' => $domain->getDomainId(),
+        '!e' => $error,
+      ]));
+    }
+    else {
+      try {
+        $domain->save();
+      }
+      catch (EntityStorageException $e) {
+        throw new DomainCommandException('Unable to save domain', $e);
+      }
+
+      if ($domain->getDomainId()) {
+        $this->logger()->info(dt('Created @name at @domain.',
+          ['@name' => $domain->label(), '@domain' => $domain->getHostname()]));
+      }
+      else {
+        $this->logger()->error(dt('The request could not be completed.'));
+      }
+    }
+  }
+
+  /**
+   * Validates a domain.
+   *
+   * @param $hostname
+   *   The domain name to validate for syntax and uniqueness.
+   *
+   * @return array
+   *   An array of errors encountered.
+   *
+   * @see domain_validate()
+   */
+  protected function validateDomain($hostname) {
+    /** @var \Drupal\domain\DomainValidatorInterface $validator */
+    $validator = \Drupal::service('domain.validator');
+    return $validator->validate($hostname);
+  }
+
+  /**
+   * Deletes a domain record.
+   *
+   * @param DomainInterface[] $domains
+   *   The domain_id to delete. Pass 'all' to delete all records.
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   * @throws \UnexpectedValueException
+   */
+  protected function deleteDomain($domains) {
+    foreach ($domains as $domain) {
+      if (!$domain instanceof DomainInterface) {
+        throw new StorageException('deleting domains: value is not a domain');
+      }
+      $hostname = $domain->getHostname();
+
+      if ($this->is_dry_run) {
+        $this->logger()->info(dt('DRYRUN: Domain record @domain deleted.',
+          ['@domain' => $hostname]));
+        continue;
+      }
+
+      try {
+        $domain->delete();
+      }
+      catch (EntityStorageException $e) {
+        throw new DomainCommandException(dt('Unable to delete domain: @domain',
+          ['@domain' => $hostname]), $e);
+      }
+      $this->logger()->info(dt('Domain record @domain deleted.',
+        ['@domain' => $hostname]));
+    }
+  }
+
+  /**
+   * Determine a list of the entities that are domain enabled.
+   *
+   * @return string[]
+   *   List of entities supporting domain access.
+   */
+  protected function findDomainEnabledEntities(string $using_field = DOMAIN_ACCESS_FIELD) {
+    $this->ensureEntityFieldMap();
+    $entities = [];
+    foreach($this->entity_field_map as $type => $fields) {
+      if (array_key_exists($using_field, $fields)) {
+        $entities[] = $type;
+      }
+    }
+    return $entities;
+  }
+
+  /**
+   * Determine whether or not a given entity is domain-enabled.
+   *
+   * @return string[]
+   *   List of entities supporting domain access.
+   */
+  protected function entityHasDomainField(string $entity_type, string $field = DOMAIN_ACCESS_FIELD) {
```

Comment:
This might become a public method on the DomainElementManager.

I can see a use-case for that.